ui/apps/platform/src/Components/TypeaheadSelect/TypeaheadSelect.tsx

```diff
@@ -0,0 +1,195 @@
+    id: string;
+    value: string;
+    onChange: (value: string) => void;
+    options: TypeaheadSelectOption[];
+    allowCreate?: boolean;
+    placeholder?: string;
+    isDisabled?: boolean;
+    toggleAriaLabel?: string;
+    onBlur?: React.FocusEventHandler<HTMLDivElement>;
+    menuAppendTo?: () => HTMLElement;
+    footer?: React.ReactNode;
+    maxHeight?: string;
+    direction?: 'up' | 'down';
+};
+
+function TypeaheadSelect({
+    id,
+    value,
+    onChange,
+    options,
+    allowCreate = false,
+    placeholder = 'Type to search...',
+    isDisabled = false,
+    toggleAriaLabel,
+    onBlur,
+    menuAppendTo,
+    footer,
+    maxHeight = '300px',
+    direction = 'down',
+}: TypeaheadSelectProps): ReactElement {
+    const [isOpen, setIsOpen] = useState(false);
+    const [inputValue, setInputValue] = useState('');
+    const [focusedItemIndex, setFocusedItemIndex] = useState<number>(-1);
+
+    function onSelect(
+        _event: React.MouseEvent<Element, MouseEvent> | undefined,
+        selection: string | number | undefined
+    ) {
+        if (typeof selection === 'string') {
+            setIsOpen(false);
+            onChange(selection);
+            setInputValue('');
+        }
+    }
+
+    function onToggle() {
+        setIsOpen(!isOpen);
+    }
+
+    function onInputChange(_event: React.FormEvent<HTMLInputElement>, text: string) {
+        setInputValue(text);
+        setFocusedItemIndex(-1); // Reset focus when typing
+        if (!isOpen) {
+            setIsOpen(true);
+        }
+    }
+
+    function onKeyDown(event: React.KeyboardEvent) {
+        const allOptions = shouldShowCreateOption
+            ? [...filteredOptions, { value: inputValue, label: `Create "${inputValue}"` }]
+            : filteredOptions;
+
+        if (event.key === 'Escape') {
+            setIsOpen(false);
+            setInputValue('');
+            setFocusedItemIndex(-1);
+        } else if (event.key === 'Enter' && isOpen) {
+            event.preventDefault();
+            if (focusedItemIndex >= 0 && focusedItemIndex < allOptions.length) {
+                const selectedOption = allOptions[focusedItemIndex];
+                onChange(selectedOption.value);
+                setIsOpen(false);
+                setInputValue('');
+                setFocusedItemIndex(-1);
+            }
+        } else if (event.key === 'ArrowDown' && isOpen) {
+            event.preventDefault();
+            setFocusedItemIndex((prev) => (prev < allOptions.length - 1 ? prev + 1 : 0));
+        } else if (event.key === 'ArrowUp' && isOpen) {
+            event.preventDefault();
+            setFocusedItemIndex((prev) => (prev > 0 ? prev - 1 : allOptions.length - 1));
+        }
+    }
+
+    // Filter options based on input
+    const filteredOptions = inputValue
+        ? options.filter((option) => {
+              const label = option.label || option.value;
+              return label.includes(inputValue);
```

Comment:
What do you think about:

```suggestion
              return label.toLowerCase().includes(inputValue.toLowerCase());
```